forum/qa-content/qa-ask.js

```diff
@@ -1,300 +1,300 @@
 {
-	var startval=startpath ? startpath.split("/") : [];
-	var setdescnow=true;
+    let startval=startpath ? startpath.split("/") : [];
+    let setdescnow=true;
 
-	for (var l=0; l<=qa_cat_maxdepth; l++) {
-		var elem=document.getElementById(idprefix+'_'+l);
+    for (let l=0; l<=qa_cat_maxdepth; l++) {
+        let elem=document.getElementById(idprefix+'_'+l);
 
-		if (elem) {
-			if (l) {
-				if (l<startval.length && startval[l].length) {
-					var val=startval[l];
+        if (elem) {
+            if (l) {
+                if (l<startval.length && startval[l].length) {
+                    let val=startval[l];
 
-					for (var j=0; j<elem.options.length; j++)
-						if (elem.options[j].value==val)
-							elem.selectedIndex=j;
-				} else
-					var val=elem.options[elem.selectedIndex].value;
-			} else
-				val='';
+                    for (let j=0; j<elem.options.length; j++)
+                        if (elem.options[j].value==val)
+                            elem.selectedIndex=j;
+                } else
+                    let val=elem.options[elem.selectedIndex].value;
+            } else
+                val='';
 
-			if (elem.qa_last_sel!==val) {
-				elem.qa_last_sel=val;
+            if (elem.qa_last_sel!==val) {
+                elem.qa_last_sel=val;
 
-				var subelem=document.getElementById(idprefix+'_'+l+'_sub');
-				if (subelem)
-					subelem.parentNode.removeChild(subelem);
+                let subelem=document.getElementById(idprefix+'_'+l+'_sub');
+                if (subelem)
+                    subelem.parentNode.removeChild(subelem);
 
-				if (val.length || (l==0)) {
-					subelem=elem.parentNode.insertBefore(document.createElement('span'), elem.nextSibling);
-					subelem.id=idprefix+'_'+l+'_sub';
-					qa_show_waiting_after(subelem, true);
+                if (val.length || (l==0)) {
+                    subelem=elem.parentNode.insertBefore(document.createElement('span'), elem.nextSibling);
+                    subelem.id=idprefix+'_'+l+'_sub';
+                    qa_show_waiting_after(subelem, true);
 
-					qa_ajax_post('category', {categoryid:val},
-						(function(elem, l) {
-							return function(lines) {
-								var subelem=document.getElementById(idprefix+'_'+l+'_sub');
-								if (subelem)
-									subelem.parentNode.removeChild(subelem);
+                    qa_ajax_post('category', {categoryid:val},
+                        (function(elem, l) {
+                            return function(lines) {
+                                let subelem=document.getElementById(idprefix+'_'+l+'_sub');
+                                if (subelem)
+                                    subelem.parentNode.removeChild(subelem);
 
-								if (lines[0]=='1') {
-									elem.qa_cat_desc=lines[1];
+                                if (lines[0]=='1') {
+                                    elem.qa_cat_desc=lines[1];
 
-									var addedoption=false;
+                                    const addedoption=false;

```

Comment:
Z tego co widzę, to do zmiennej **addedoption** (swoją drogą, jeśli już zadajesz sobie trud przejrzenia wszystkich zmiennych, to można też "przepisać" te, które się do tego nadają, na *camelCase*) jest później przypisywany true (@246), więc nie możesz tutaj dać `const`, bo to słówko uniemożliwia późniejsze przypisanie innej wartości do zmiennej - tutaj powinieneś użyć `let`. Biorąc to pod uwagę dobrze by było, gdybyś upewnił się, że wszędzie gdzie zamieniłeś `var` na `const` do zmiennej nie jest przypisywana już inna wartość.